src/main/kotlin/translations/intentions/ConvertToTranslationIntention.kt

```diff
@@ -73,12 +77,19 @@ class ConvertToTranslationIntention : PsiElementBaseIntentionAction() {
             val key = result.first ?: return
             val replaceLiteral = result.second
             try {
-                TranslationFiles.add(element, key, value)
+                if (existingKey != key) {
+                    TranslationFiles.add(element, key, value)
+                }
                 if (replaceLiteral) {
+                    val translationSettings = TranslationSettings.getInstance(project)
                     val psi = PsiDocumentManager.getInstance(project).getPsiFile(editor.document) ?: return
                     psi.runWriteAction {
                         val expression = JavaPsiFacade.getElementFactory(project).createExpressionFromText(
-                            "net.minecraft.client.resources.I18n.format(\"$key\")",
+                            if (translationSettings.isUseCustomConvertToTranslationTemplate) {
+                                translationSettings.convertToTranslationTemplate.replace("\$key", key)
+                            } else {
+                                "net.minecraft.client.resources.I18n.format(\"$key\")"
```

Comment:
Wasn't the point of adding a checkbox for a custom template to use the mappings manager?

Reply:
My understanding from your previous message is that the mappings API isn't ready yet.
I added the checkbox so that when it's ready it can be integrated without requiring users to tick another box when they update.